packages/telegram-cloud-storage-stamper/src/index.ts

```diff
@@ -0,0 +1,300 @@
+  async setSigningKey(config?: TTelegramCloudStorageStamperConfig) {
+    if (!config) {
+      // try to get API key at default location and set the stamper to use that
+      // passing the DEFAULT_TURNKEY_CLOUD_STORAGE_KEY is not necessary since getAPIKey will use that by default
+      let { apiPublicKey, apiPrivateKey } = await this.getAPIKey(
+        DEFAULT_TURNKEY_CLOUD_STORAGE_KEY
+      );
+
+      if (apiPublicKey && apiPrivateKey) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey,
+          apiPrivateKey,
+        });
+        return true;
+      }
+    } else if (config.cloudStorageKey && !config.cloudStorageAPIKey) {
+      // try to get API key at specified location and set the stamper to use that
+      let { apiPublicKey, apiPrivateKey } = await this.getAPIKey(
+        config.cloudStorageKey
+      );
+
+      if (apiPublicKey && apiPrivateKey) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey,
+          apiPrivateKey,
+        });
+        return true;
+      }
+    } else if (!config.cloudStorageKey && config.cloudStorageAPIKey) {
+      // try to set API key at default location and set the stamper to use that
+      // passing the DEFAULT_TURNKEY_CLOUD_STORAGE_KEY is not necessary since insertAPIKey will use that by default
+      let stored = await this.insertAPIKey(
+        config.cloudStorageAPIKey.apiPublicKey,
+        config.cloudStorageAPIKey.apiPrivateKey,
+        DEFAULT_TURNKEY_CLOUD_STORAGE_KEY
+      );
+
+      if (stored === true) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey: config.cloudStorageAPIKey.apiPublicKey,
+          apiPrivateKey: config.cloudStorageAPIKey.apiPrivateKey,
+        });
+        return true;
+      }
+    } else if (config.cloudStorageKey && config.cloudStorageAPIKey) {
+      // try to set API key at specified location and set the stamper to use that
+      let stored = await this.insertAPIKey(
+        config.cloudStorageAPIKey.apiPublicKey,
+        config.cloudStorageAPIKey.apiPrivateKey,
+        config.cloudStorageKey
+      );
+
+      if (stored === true) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey: config.cloudStorageAPIKey.apiPublicKey,
+          apiPrivateKey: config.cloudStorageAPIKey.apiPrivateKey,
+        });
+        return true;
+      }
+    }
+    return false;
+  }
+
+  async insertItem(key: string, value: string) {
+    // check to see if we're in a telegram mini app context
+    this.checkTelegramContext();
+
+    try {
+      return await this.setCloudStorageItem(key, value);
+    } catch {
+      return false;
+    }
+  }
+
+  async insertAPIKey(
+    apiPublicKey: string,
+    apiPrivateKey: string,
+    key: string = DEFAULT_TURNKEY_CLOUD_STORAGE_KEY
+  ) {
+    try {
+      return await this.insertItem(
+        key,
+        this.stringifyAPIKey(apiPublicKey, apiPrivateKey)
+      );
+    } catch {
+      return false;
+    }
+  }
```

Comment:
same here. I think the interface should just be an undefined return value rather than an undocumented boolean. Let's throw if something goes wrong.

I actually think this function can potentially stop existing because it'd just be:
```ts
async insertAPIKey(
    apiPublicKey: string,
    apiPrivateKey: string,
    key: string = DEFAULT_TURNKEY_CLOUD_STORAGE_KEY
  ) {
    return await this.insertItem(
        key,
        this.stringifyAPIKey(apiPublicKey, apiPrivateKey)
      );
}
```

I think the caller can call `this.InsertItem` themselves? I don't feel super strongly either way.

Reply:
Basically, if this function didn't exist, we'd go from:
```ts
let stored = await this.insertAPIKey(
    config.cloudStorageAPIKey.apiPublicKey,
    config.cloudStorageAPIKey.apiPrivateKey,
    DEFAULT_TURNKEY_CLOUD_STORAGE_KEY
);
```

to
```ts
this.insertItem(
    DEFAULT_TURNKEY_CLOUD_STORAGE_KEY,
    this.stringifyAPIKey(config.cloudStorageAPIKey.apiPublicKey, config.cloudStorageAPIKey.apiPrivateKey)
);
```

I actually like the version with `this.insertAPIKey` so maybe let's leave this function in! Cheap enough to keep :)